dbmongo/lib/marshal/filter.go

```diff
@@ -27,7 +27,12 @@ func IsFiltered(id string, filter map[string]bool) (bool, error) {
 	if filter == nil {
 		return false, nil
 	}
-	return !filter[id[0:9]], nil
+	return !FilterHas(id, filter), nil
+}
+
+// FilterHas tells if the siret/siren is listed in the filter, or not.
+func FilterHas(siretOrSiren string, filter map[string]bool) bool {
+	return filter[siretOrSiren[0:9]]
```

Comment:
En voyant passer cette expression je me dis que ça pourrait faire paniquer le process si par hasard on se retrouve avec une chaine de moins de 9 caractères… Peut être qu'il y a déjà des gardes fous avant l'appel de cette fonction ?
Sinon à défaut de mettre un garde fou en amont, il faudrait au moins faire quelque chose de ce genre:
```suggestion
	if len(sirenOrSiren) >= 9 {
                return filter[siretOrSiren[0:9]]
	}
	return false
```

Le risque a l'air faible parce que le code précédent était déjà à risque et on a jamais constaté de panic, probablement grâce à la stabilité des fichiers d'input sur ce point, ce qui veut aussi dire que c'est sans doute un peu risqué de compter là dessus.

Reply:
Bien vu !